### PR TITLE
Conducting Good Retrospectives

### PR DESCRIPTION
**Pull request by mhanberg**

### CODE REVIEW

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+title: Conducting Good Retrospectives
+date: 2018-10-30 09:00:00 -04:00
+categories: post
+desc: Introduction to holding wildly productive and efficient team retrospectives.
+permalink: /:categories/:year/:month/:day/:title/
+image: https://res.cloudinary.com/mhanberg/image/upload/t_scale/v1541683676/jametlene-reskp-651411-unsplash.jpg
+image_desc: Photo by Jametlene Reskp on Unsplash
+---
+
+Your team has probably experienced moments of success or failure, moments of working like a well-oiled-machine or not being able to stand the sound of each other's voices. What you've probably also done is pass these moments by without taking time to ponder "Why did these things happen"?
```
Comment (mike-rogers) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: `well oiled machine` - no hyphens necessary

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+This is when your team holds a __retrospective__.
+
+### What is a retrospective?
+
+A retrospective (or "retro") is a recurring meeting used by teams to look back at the last week or two in order to figure out what went well and what went poorly. The realm of topics is wide ranging: from missing deadlines to mishandled social interactions. These meetings aren't just to make you feel good or bad about yourselves, they're designed to curate items on which your team will take action after the meeting.
+
+Setting aside dedicated time for these matters assures that they will get some attention and helps keep these conversations from spilling into normal work time (this isn't to say that you shouldn't discuss something when it happens).
+
+Along with **learning from your successes and failures**, one of the goals of a retrospective to **build trust amongst your team**. Teams perform better when they operate in an environment of [psychological safety](https://hbr.org/2017/08/high-performing-teams-need-psychological-safety-heres-how-to-create-it).
```
Comment (mike-rogers) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: I like the trust-building angle but I didn't see anything later that emphasized or exemplified the idea.

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+#### Facilitation
+
+Someone should volunteer to facilitate the meeting. This person will lead the chosen activity for the meeting. Depending on the activity, this can include: keeping a conversation timer, taking notes, or tracking whose turn it is to speak. 
+
+Having a facilitator **isolates the responsibility of running the meeting to one person**, allowing everyone else to concentrate on participating.
+
+#### Conversational Turn Taking
+
+It's very important that everyone can be heard. Your team should decide on a signal that makes it clear you have something to say next. My team has opted to use a finger raise when you want to say something. Once the person currently speaking is done, they or the facilitator would motion to the next person in line to speak.
```
Comment (mike-rogers) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: You could link to Google's Project Aristotle, which found that psychological safety and conversational turn-taking were the two factors most associated with performing teams.

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+1. Vote on topics
+1. Discuss topics
+    - Limit to 5 minutes, then vote on continuing with a 2 minute extension, or else move on to next topic.
+    - Facilitator will take notes and track action items.
+
+Any topics that aren't discussed are carried over to the next meeting, unless the author states otherwise.
+
+#### Sprint Retrospective
+
+During the Sprint Retrospective we use the [Four Quadrant](http://www.funretrospectives.com/lessons-learned-quadrants-planning-vs-success/) activity. As with the Mid-Sprint retro, the chosen facilitator will explain the activity and pass out the required supplies.
```
Comment (mike-rogers) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: The grid that we do isn't exactly the Four Quadrant activity. The axes are different and the execution is different. We can chat in-person if you'd like.

When I first ran the grid thing, part of the activity was supposed to be the group collaborating on where to put the various event-stickies on a given axis. I think that part has been lost over the last couple of years. I wonder if it was worthwhile...

Incidentally this paragraph gave me a neat idea for a different retro activity. Remind me to talk to you later. :smile:

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+**Escalate** - bad and out of your control: these events can only be addressed by someone external to your team. Now is the time to figure out with who to escalate the issue, or assign someone the action of figuring that out later.
+
+**Kudos** - good and out of your control: these are positive events that were caused by someone external to your team. Here the team will assign an action to someone to reach out and express the team's gratitude.
+
+## Conclusion
+
+Done right, retrospectives can be a boon for successful and struggling teams alike. 
+
+The goal is to **build a culture of trust** amongst your team, where everyone has the freedom to express themselves without the fear of being punished for making a mistake or speaking their mind. If there are barriers to bringing up problems your team is having, you can never expect to be able to work through them and improve.
```
Comment (mike-rogers) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: `Amongst`? Using arcane language isn't behooving.

Comment (mhanberg) [reply _posts/2018-10-30-conducting-good-retrospectives.md]: I didn't realize amongst was an old word lol.

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+Your team has probably experienced moments of success or failure, moments of working like a well-oiled-machine or not being able to stand the sound of each other's voices. What you've probably also done is pass these moments by without taking time to ponder "Why did these things happen"?
+
+This is when your team holds a __retrospective__.
+
+### What is a retrospective?
+
+A retrospective (or "retro") is a recurring meeting used by teams to look back at the last week or two in order to figure out what went well and what went poorly. The realm of topics is wide ranging: from missing deadlines to mishandled social interactions. These meetings aren't just to make you feel good or bad about yourselves, they're designed to curate items on which your team will take action after the meeting.
+
+Setting aside dedicated time for these matters assures that they will get some attention and helps keep these conversations from spilling into normal work time (this isn't to say that you shouldn't discuss something when it happens).
```
Comment (rpherbig) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: The combination of sentence length + parenthetical phrasing makes it a bit hard to parse.

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+Along with **learning from your successes and failures**, one of the goals of a retrospective to **build trust amongst your team**. Teams perform better when they operate in an environment of [psychological safety](https://hbr.org/2017/08/high-performing-teams-need-psychological-safety-heres-how-to-create-it).
+
+## How to conduct a retrospective
+
+Anyone can host a retrospective, but hosting a _good_ retrospective takes some finesse. There are four key aspects that I believe are paramount to a good retrospective: Facilitation, Conversational Turn Taking, Format/Activity, and Action Items.
+
+#### Facilitation
+
+Someone should volunteer to facilitate the meeting. This person will lead the chosen activity for the meeting. Depending on the activity, this can include: keeping a conversation timer, taking notes, or tracking whose turn it is to speak.
```
Comment (rpherbig) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: Is it worth noting that these activities can be done by different people - or does that just muddy the waters?

File: _posts/2018-10-30-conducting-good-retrospectives.md (hunk 4 above)
Comment (mhanberg) [reply _posts/2018-10-30-conducting-good-retrospectives.md]: Interesting, I did notice that the activity described in that link wasn't exactly like how I described it, but I figured it was similar enough to keep the link. 

I think I'll add a bit about how this is a slightly different take on that activity

File: _posts/2018-10-30-conducting-good-retrospectives.md (hunk 3 above)
Comment (mhanberg) [reply _posts/2018-10-30-conducting-good-retrospectives.md]: I'll check that out!

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+Often one might have something to say that is tangential to the current conversation, and will delay their turn to speak to let the current thread of discussion finish.
+
+The primary motivation for the technique to is **avoid interruptions** and **talking over each other**.
+
+#### Format/Activity
+
+Free form conversation is not conducive to reaching conclusions, so it helps to pick a meeting format that fits well with your team. At the beginning of the meeting, the facilitator will explain the format and confirm that everyone is okay with that decision. Different formats optimize for different aspects of your team: project performance, interpersonal communication, or general frustration.
+
+Picking a format gives the **meeting structure** and often an **objective**, allowing the team to work together towards a known and common goal.
```
Comment (rpherbig) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: Maybe `meeting **structure**`?

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+During the Mid-Sprint Retrospective, we engage in the [Lean Coffee](http://leancoffee.org) activity (unless someone suggests an alternate activity!). The chosen facilitator will explain the chosen activity and pass out the required supplies. 
+
+The basics of the Lean Coffee activity are:
+
+1. Generate discussion topics
+    - Limit to 5 minutes
+1. Vote on topics
+1. Discuss topics
+    - Limit to 5 minutes, then vote on continuing with a 2 minute extension, or else move on to next topic.
```
Comment (rpherbig) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: Is it worth noting that we tinkered with the 5/2 durations - or does that just muddy the waters?

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+    - Facilitator will take notes and track action items.
+
+Any topics that aren't discussed are carried over to the next meeting, unless the author states otherwise.
```
Comment (rpherbig) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: Is it worth noting that we allowed anyone to keep a topic alive?

File: _posts/2018-10-30-conducting-good-retrospectives.md (hunk 4 above)
Comment (rpherbig) [reply _posts/2018-10-30-conducting-good-retrospectives.md]: > the group collaborating on where to put the various event-stickies on a given axis

It's a pretty time consuming part of the activity, and most people seemed to feel it didn't add much value.

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+
+## Case Study
+
+For the last year, my team has had a weekly retro, alternating the format each week. We operate in two week sprints, so in the middle of the sprint we typically concentrate on interpersonal issues or other general frustrations and at the end of the sprint we focus on project-specific issues.
+
+However, at the beginning of any retrospective, we solicit the opportunity to suggest an alternative format.
+
+#### Mid-Sprint Retrospective
+
+During the Mid-Sprint Retrospective, we engage in the [Lean Coffee](http://leancoffee.org) activity (unless someone suggests an alternate activity!). The chosen facilitator will explain the chosen activity and pass out the required supplies.
```
Comment (rpherbig) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: Verb-tense is back-and-forth throughout the post: 'engage' or 'engaged'?

File: _posts/2018-10-30-conducting-good-retrospectives.md (hunk 4 above)
Comment (rpherbig) [reply _posts/2018-10-30-conducting-good-retrospectives.md]: https://medium.com/serious-scrum/adding-another-dimension-to-the-traditional-scrum-retrospective-format-6014d89ebebb

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+Starting in the top right corner and moving counter clockwise, the team discusses the events in an attempt to learn. If your team is prone to drawing out discussions, consider using a Lean Coffee style discussion timer.
+
+**Keep doing** - good and in your control: give yourselves a quick pat on the back to celebrate good work.
+
+**Improve** - bad and in your control: these are events that your team has the power to avoid in the future. The facilitator will lead discussion on the topics, driving toward actions that can be taken in the future.
+
+**Escalate** - bad and out of your control: these events can only be addressed by someone external to your team. Now is the time to figure out with who to escalate the issue, or assign someone the action of figuring that out later.
+
+**Kudos** - good and out of your control: these are positive events that were caused by someone external to your team. Here the team will assign an action to someone to reach out and express the team's gratitude.
+
```
Comment (rpherbig) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: I feel like a sticky being "in the middle" or "on the line" is common enough that it deserves a mention and explanation. What do you think?

File: _posts/2018-10-30-conducting-good-retrospectives.md [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+1. The facilitator then draws a vertical axis, with the top indicating the event was _in the team's control_ and the bottom that it was _not in the team's control_.
+1. The facilitator quickly pulls each sticky to the appropriate place on the vertical axis. This is usually fairly obvious, but the rest of the team will signal where they think it should go in case of disagreement.
+
+At this point, the board should resemble something like this:
+
+![Diagram of a completed Four Quadrants diagram]({{ 'https://res.cloudinary.com/mhanberg/image/upload/v1541686229/four-quadrants.png' }})
+
+Starting in the top right corner and moving counter clockwise, the team discusses the events in an attempt to learn. If your team is prone to drawing out discussions, consider using a Lean Coffee style discussion timer.
+
+**Keep doing** - good and in your control: give yourselves a quick pat on the back to celebrate good work.
```
Comment (rpherbig) [inline _posts/2018-10-30-conducting-good-retrospectives.md]: Potentially also discuss how to enable the team to continue these things (if, say, a stick was about an experiment).
